build: introduce and switch to GYB_SOURCES

compnerd opened the pull request:
This avoids us having to pattern match every source file which should
help speed up the CMake generation.  A secondary optimization is
possible with CMake 3.14 which has the ability to remove the last
extension component without having to resort to regular expressions.  It
also helps easily identify the GYB'ed sources.

<!-- What's in this pull request? -->
Replace this paragraph with a description of your changes and rationale. Provide links to external references/discussions if appropriate.

<!-- If this pull request resolves any bugs in the Swift bug tracker, provide a link: -->
Resolves [SR-NNNN](https://bugs.swift.org/browse/SR-NNNN).

<!--
Before merging this pull request, you must run the Swift continuous integration tests.
For information about triggering CI builds via @swift-ci, see:
https://github.com/apple/swift/blob/master/docs/ContinuousIntegration.md#swift-ci

Thank you for your contribution to Swift!
-->

compnerd commented:
@swift-ci please test

compnerd commented:
CC: @gottesmm @Rostepher @gwynne

gottesmm commented:
I think this change makes sense at a high level. A few requests:

1. Can you put a newline in between GYB_SOURCES and the previous file. This will help to reduce merge conflicts IIUC if someone adds more files out of tree that are not gybed. Otherwise, we will continually hit merge conflicts I think.
2. Can you split the commit into two commits (not separate PRs), one introducing the change and the other applying it. Would just make it easier to review.

compnerd committed:
build: introduce and switch to GYB_SOURCES

This avoids us having to pattern match every source file which should
help speed up the CMake generation.  A secondary optimization is
possible with CMake 3.14 which has the ability to remove the last
extension component without having to resort to regular expressions.  It
also helps easily identify the GYB'ed sources.

compnerd commented:
@swift-ci please test

compnerd commented:
@swift-ci please test macOS platform

compnerd commented:
@swift-ci please test macOS platform

compnerd commented:
@swift-ci please test

compnerd commented:
@swift-ci please test

compnerd commented:
@swift-ci please test

compnerd commented:
@swift-ci please test

compnerd commented:
@swift-ci please test

compnerd commented:
@swift-ci please test

gottesmm commented:
LGTM.
